savant/client/runner/source.py

```diff
@@ -205,6 +199,34 @@ def send_shutdown(self, source_id: str, auth: str) -> SourceResult:
 
         return result
 
+    def _send_frame(self, source: Frame) -> SourceResult:
+        """Send a single frame to ZeroMQ socket.
+
+        :param source: Source of the frame to send. Can be an instance
+            of FrameSource or a tuple of VideoFrame and content.
+        :return: Result of sending the frame.
+        """
+
+        zmq_topic, message, content, result = self._prepare_video_frame(source)
+        self._send_zmq_message(zmq_topic, message, content)
+        logger.debug('Sent video frame %s/%s.', result.source_id, result.pts)
+
+        return result
+
+    def _send_batch(self, source_id: str, source: Batch) -> SourceResult:
```

Comment:
No need to pass `source_id` separately, you can get `source_id` from `VideoFrameBatch.frames`.

Reply:
Batch should be related to a single topic. Otherwise, there's no point in using a batch, as we would need to unpack it to send frames based on their source (don't consider sending the same batch to all topics because it's pretty strange)